firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/send/ReportQueue.java

```diff
@@ -115,6 +117,24 @@ TaskCompletionSource<CrashlyticsReportWithSessionId> enqueueReport(
     }
   }
 
+  public void flushScheduledReportsIfAble() {
+    CountDownLatch latch = new CountDownLatch(1);
+    new Thread(
+            () -> {
+              TransportRuntime.getInstance()
+                  .getUploader()
+                  .upload(DataTransportCrashlyticsReportSender.transportContext, 1, () -> {});
+              latch.countDown();
+            })
+        .start();
+    try {
+      //noinspection ResultOfMethodCallIgnored best effort only
+      latch.await(2, TimeUnit.SECONDS);
```

Comment:
If you block waiting for the thread to finish, why have a `Thread` at all instead of calling the blocking method directly?

Reply:
We want this to time out after 2 seconds, so it never leaves the app hanging after the uncaught exception.